src/payment/models/buy/buy.service.ts

```diff
@@ -63,27 +65,48 @@ export class BuyService {
       .then((r) => r.volume);
   }
 
-  // --- BUYS --- //
-  async createBuy(userId: number, userAddress: string, dto: CreateBuyDto): Promise<Buy> {
-    // check asset
-    const asset =
-      dto.type === BuyType.WALLET
-        ? await this.assetService.getAsset(dto.asset.id)
-        : await this.assetService.getAssetByDexName('DFI');
-    if (!asset) throw new BadRequestException('Asset not found');
+  // --- PAYMENT-INFO --- //
+  async getBuyPaymentInfos(
```

Comment:
I think it would be easier to call the complete `createBuy` method here and catch the ConflictException (or add a parameter to ignore conflict).